File: insertionsort.c
#include<stdio.h>
void main()
{
    int number[]={5,3,8,4,2};
    int n=5;
    for(int i=1;i<n;i++)
    {
        int temp=number[i];
        int j=i-1;
        while(j>=0 && temp<number[j])
        {
            number[j+1]=number[j];
            j=j-1;
        }
        number[j+1]=temp;
    }
    for(int i=0;i<n;i++)
    {
        printf("%d",number[i]);
    }
}

File: Insertion_Sort.java

public class Insertion_Sort {
	public static void main(String args[])
	{
		int n=5,temp=0,j;
		int a[] = {34,67,23,657,4};
		for(int i=1;i<n;i++)	// i starts from 1 because the array should start from the second position
		{
			temp=a[i];			
			j=i-1;				// j is assigned with the array index before i'th index 
			while(j>=0 && a[j]>temp)  //checking if j > 0 and a[j] element is greater than the element in i'th position.
			{
				a[j+1]=a[j];		//swap the elements
				j--;				//we need to traverse in reverse on the sorted part of array
			}
			a[j+1]=temp;			// j+1 because j-- will take place before the while loop is terminated
		}
		for(int i:a)
		System.out.print(i+" ");
	}

}
